src/Common/Provider/Controller.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * The base clas all Controllers should extend.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Provider;
+ */
+
+namespace TEC\Common\Provider;
+
+use tad_DI52_ServiceProvider as Service_Provider;
+use TEC\Common\StellarWP\ContainerContract\ContainerInterface;
+use Tribe__Log as Log;
+
+/**
+ * Class Controller.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Provider;
+ *
+ * @property ContainerInterface $container
+ */
+abstract class Controller extends Service_Provider {
+	/**
+	 * Registers the filters and actions hooks added by the controller if the controller has not registered yet.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register() {
+		/*
+		 * Look up and set the value in the container request cache to allow building the same Controller
+		 * with a **different** container. (e.g. in tests).
+		 */
+		if ( $this->container->getVar( static::class . '_registered' ) ) {
```

Comment:
Should we set '_registered' as a prop? Maybe one that is not editable for sanity's sake?

Reply:
Having it in our global vars can be useful to check outside, as we sometimes do. And I'd have one source of truth.

Reply:
Although I prefer something like `tribe(Some_Controller::class)->is_active()` which references that var...

Reply:
> Although I prefer something like tribe(Some_Controller::class)->is_active()
That is already part of the API, but it's about the Controller being "active" as in "relevant for the current request"; e.g. a wp-cli controller is not active when the request context is not the a wp-cli one. I guess you mean a method like `Controller::is_registered`, which is currently lacking from the API. I've added it. 
Note: I've updated the `Controller_Test_Case` to replace the global service locator (`tribe()`) with the test one when using `Controller_Test_Case::make_controller` to make sure the `Controller::is_registered` method is correct spoofed.